### ISSUE TITLE
[BUG] Watermark not effective

### ISSUE THREAD

**Issue by PhotosESM** [Piwigo/Piwigo-Mobile#370]
When the watermark is active on the site it doesn't works well in iOS.

First the watermark appears when the photo is loading (and in low quality) but after, when the high quality photo is shown in the App there is no more Watermark.

**Comment by EddyLB** [comment]
With the default configuration of a Piwigo server, there is no watermark on photos of width or height is greater than 500 pixels. So there is not watermark of the following sizes: _Square_, _Thumbnail_, _XXS - tiny_, _XS - extra small_. And there is a watermark on photos of the other sizes: _S - small_, _Medium_, _L - large_, _XL - extra large_ and _XXL - Huge_. You can easily check that by changing the size of the thumbnail files of albums and photos in the Preferences of the iOS app.

If you change the settings of your Piwigo server in Configuration / Watermark so that a watermark is applied on smaller photos, say for example width or height greater than 100 pixels, all photo will have a watermark ;-)

However… there is no watermark on full resolution images. So if your select the full resolution for previewing photos on an iPad for example, the original photo is presented without watermark and can be shared, saved, etc. I do not know if the [Media Pump](https://piwigo.org/ext/extension_view.php?eid=828) extension could add a watermark for you. You could try or better ask on the forum.

In short, this is not a bug in the iOS app and I therefore close the issue.

**Comment by PhotosESM** [comment]
﻿Hi,

Ok thank you.

All the conf was OK.

The problem was that the Highres format was active and not watermarked, thanks for the idea at the bottom of the mail, I disabled it in the admin and it’s now ok in the app.


Envoyé depuis mon iPhone,

Cédric Barbin
+33 6 753 53000





Cordialement,

Cédric Barbin
Consultant transformation digitale / DPO Externe
+33 6 753 53000
Le 1 avr. 2020 à 22:43, Eddy Lelièvre-Berna <notifications@github.com> a écrit :

﻿

With the default configuration of a Piwigo server, there is no watermark on photos of width or height is greater than 500 pixels. So there is not watermark of the following sizes: Square, Thumbnail, XXS - tiny, XS - extra small. And there is a watermark on photos of the other sizes: S - small, Medium, L - large, XL - extra large and XXL - Huge. You can easily check that by changing the size of the thumbnail files of albums and photos in the Preferences of the iOS app.

If you change the settings of your Piwigo server in Configuration / Watermark so that a watermark is applied on smaller photos, say for example width or height greater than 100 pixels, all photo will have a watermark ;-)

However… there is no watermark on full resolution images. So if your select the full resolution for previewing photos on an iPad for example, the original photo is presented without watermark and can be shared, saved, etc. I do not know if the Media Pump<https://piwigo.org/ext/extension_view.php?eid=828> extension could add a watermark for you. You could try or better ask on the forum.

In short, this is not a bug in the iOS app and I therefore close the issue.

**Comment by PhotosESM** [comment]
Ok thank you.

All the conf was OK.

The problem was that the Highres format was active and not watermarked, thanks for the idea at the bottom of the mail, I disabled it in the admin and it’s now ok in the app.

**Comment by EddyLB** [comment]
I forgot that option. You can also disable the high-resolution to specific users.
